nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/providers/zookeeper/ZooKeeperStateProvider.java

```diff
@@ -147,6 +249,26 @@ public synchronized void init(final StateProviderInitializationContext context)
         }
     }
 
+    private void setPropertyIfNotNull(Properties properties, String key, String value) {
+        if(value != null) {
+            properties.setProperty(key, value);
+        }
+    }
+
+    private boolean keystorePropertiesAreValid(String keyStorePath, String keyStoreType, String keyStorePassword) {
+        if(keyStorePath != null && keyStorePath != null && keyStorePassword != null) {
+            try {
+                return KeyStoreUtils.isStoreValid(Paths.get(keyStorePath).toUri().toURL(), KeystoreType.valueOf(keyStoreType), keyStorePassword.toCharArray());
+            } catch (MalformedURLException e) {
+                logger.error("ZooKeeperStateProvider keyStorePath " + keyStorePath + " is not a valid file URL or could not be found.");
+                e.printStackTrace();
```

Comment:
The exception should be passed to logger.error() instead of writing to standard error.